### PR TITLE
feat: add home screen UI with play mode buttons

### LINKED ISSUE #14: implement a nice home screen with 2 buttons
implement a nice home screen with 2 buttons. 1 play vs stockfish and another with play vs player

### PR DESCRIPTION
**Pull request by stefangiurca1229**
## Summary
This PR introduces a visually appealing home screen as the main entry point for the application. The home screen features two buttons: "Play vs Stockfish" and "Play vs Player". Each button is wired to trigger the appropriate action or route to the corresponding game mode (either as a stub or connected to existing game logic). Basic styling has been applied to ensure a user-friendly interface.

## Changes
- Added a new main entry point that launches the home screen UI
- Implemented a home screen with two buttons: "Play vs Stockfish" and "Play vs Player"
- Button actions are connected to the appropriate game modes or stubbed if not yet implemented
- Applied basic styling for improved visual appeal
- Updated documentation to explain how to launch and use the home screen

## Testing
Automated tests verify that both buttons are present, trigger the correct actions, and that the home screen launches as expected. Edge cases such as UI load failures and missing handlers are handled gracefully.

Closes #14

### CODE REVIEW

File: tests/test_home_screen.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import sys
+import pytest
+
+pytest.importorskip('tkinter')
```
Comment (stefangiurca1229) [inline tests/test_home_screen.py:12]: please add the equivalent of java doc to the tests and all methods! thank you very much

File: tests/test_home_screen.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+import tkinter as tk
+from tkinter import Button
+
+import importlib
+
+@pytest.fixture(scope="module")
+def app_module():
```
Comment (stefangiurca1229) [inline tests/test_home_screen.py:20]: please add the equivalent of java doc to the tests and all methods! thank you very much

Comment (stefangiurca1229) [reply tests/test_home_screen.py:20]: Applied — added module-level docstring and docstrings for both fixtures and all test functions in tests/test_home_screen.py to provide JavaDoc-equivalent documentation for the tests and their methods. Tests still pass (3 passed, 4 skipped).
